### PR TITLE
migrate own repo url

### PR DESCRIPTION
**Pull request by jumoog**
## Summary by Sourcery

Enhancements:
- Add functionality to migrate old repository URLs to a new URL in the plugin configuration.

### CODE REVIEW

**Comment by jumoog** [comment]
@sourcery-ai review

**Comment by jumoog** [comment]
```
[14:11:03] [INF] [1] ConfusedPolarBear.Plugin.IntroSkipper.Plugin: remove old repo url <https://raw.githubusercontent.com/intro-skipper/intro-skipper/master/manifest.json>
[14:11:03] [INF] [1] ConfusedPolarBear.Plugin.IntroSkipper.Plugin: remove old repo url <https://raw.githubusercontent.com/jumoog/intro-skipper/master/manifest.json>
[14:11:03] [INF] [1] Emby.Server.Implementations.AppBase.BaseConfigurationManager: Saving system configuration
```

**Comment by jumoog** [comment]
@sourcery-ai review

**Comment by jumoog** [comment]
@sourcery-ai review

**Review by AbandonedCart — APPROVED** [review]

File: ConfusedPolarBear.Plugin.IntroSkipper/Plugin.cs [hunk 1]
```diff
@@ -405,6 +408,62 @@ internal void CleanTimestamps(HashSet<Guid> validEpisodeIds)
...
+            {
+                foreach (var oldRepo in oldRepos)
+                {
+                    var old = pluginRepositories.Find(repo => repo.Url == oldRepo);
+                    if (old != null)
+                    {
+                        _logger.LogInformation("remove old repo url <{WebVersion}>", oldRepo);
+                        pluginRepositories.Remove(old);
+                    }
+                }
```
Comment (rlauuzo) [inline ConfusedPolarBear.Plugin.IntroSkipper/Plugin.cs:440]: pluginRepositories.RemoveAll(repo => oldRepos.Contains(repo.Url));

Comment (AbandonedCart) [reply ConfusedPolarBear.Plugin.IntroSkipper/Plugin.cs:440]: Good call. It will resolve any confusion…

**Pull request merged**